tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,389 @@
+        assert_script_run("ssh root\@$gi_vnet_routed 'ip a'", 60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+    }
+    #Destroy ROUTED NETWORK
+    assert_script_run("virsh net-destroy vnet_routed");
+    save_screenshot;
+
+##P4:ISOLATED NETWORK
+##Prepare ISOLATED NETWORK Configuration File
+    my $vnet_isolated = <<'EOF';
+<network>
+ <name>vnet_isolated</name>
+ <bridge name="virbr2"/>
+ <ip address="192.168.152.1" netmask="255.255.255.0">
+  <dhcp>
+   <range start="192.168.152.2" end="192.168.152.254" />
+  </dhcp>
+ </ip>
+</network>
+EOF
+
+    #Create ISOLATED NETWORK
+    assert_script_run("echo '$vnet_isolated' > vnet_isolated.xml");
+    assert_script_run("virsh net-create vnet_isolated.xml");
+    save_screenshot;
+    upload_logs "vnet_isolated.xml";
+    assert_script_run("rm -rf vnet_isolated.xml");
+
+    my $gi_vnet_isolated;
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "ISOLATED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_isolated --live");
+        #Get the Guest IP Address from ISOLATED NETWORK
+        if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
+            my $mac_isolated = script_output("virsh domiflist $guest | grep vnet_isolated | grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
+            script_retry "arp | grep $mac_isolated | awk \'{print \$1}\'", delay => 60, retry => 3, timeout => 60;
+            $gi_vnet_isolated = script_output("arp | grep $mac_isolated | awk \'{print \$1}\'");
+        }
+        else {
+            script_retry "journalctl | grep dnsmasq-dhcp | tail -1 | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 3, timeout => 60;
+            $gi_vnet_isolated = script_output("journalctl | grep dnsmasq-dhcp | tail -1 | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        }
+        #Confirm ISOLATED NETWORK,it is not possible to reach external network
+        assert_script_run("! ssh root\@$gi_vnet_isolated 'ping -c2 -W1 openqa.suse.de'");
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+    }
+    #Destroy ISOLATED NETWORK
+    assert_script_run("virsh net-destroy vnet_isolated");
+    save_screenshot;
+
+    #Restore default(NATed Network)
+    restore_libvirt_default;
+
+    #Restore br123 for virt_autotest
+    restore_standalone;
+
+    #Restore Guest systems
+    restore_guests;
+
+    #Restore Network setting
+    restore_network;
+
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+
+    #Restart libvirtd service
+    restart_libvirtd;
+
+    #Destroy created virtual networks
+    destroy_vir_network;
+
+    #Restore default(NATed Network)
+    restore_libvirt_default;
+
+    #Restore br123 for virt_autotest
+    restore_standalone;
+
+    #Restore Guest systems
+    restore_guests;
+
+    #Restore Network setting
+    restore_network;
+}
+
+1;
```

Comment:
Where are the necessary logs for debugging, eg host virt network config, libvirt log, etc?

Reply:
Add func upload_libvirtd_log, also each virtul network had upload virt network config.

Reply:
Done

Reply:
Yeah, you are right. Sorry for not noticing it.